src/frequenz/client/dispatch/test/_service.py

```diff
@@ -49,18 +49,56 @@ class FakeService:
     _shuffle_after_create: bool = False
     """Whether to shuffle the dispatches after creating them."""
 
+    def _check_access(self, metadata: grpc.aio.Metadata) -> None:
+        """Check if the access key is valid.
+
+        Args:
+            metadata: The metadata.
+
+        Raises:
+            grpc.RpcError: If the access key is invalid.
+        """
+        if metadata[0][0] != "key":  # type: ignore
```

Comment:
A `cast` or a `isinstance()` runtime check might be better:

1. The `ignore` is more limited and doesn't mask other typing issues
2. The `cast`ed value keeps the correct type from now on
3. Intentions are much more clear
4. In the case of `isinstance` it actually checks that the type is correct, so for example a `AttributeError` can be properly converted to an auth error if the metadata is screwed.

I'm uncertain if this ignore is supposed to affect the first `[]`, the second, or what.

Reply:
> I'm uncertain if this ignore is supposed to affect the first [], the second, or what.

same